### PR TITLE
full branch coverage by custom compiling 1.0.0

### LINKED ISSUE #2182: Bring branch coverage up to 100%
The following lines have partial coverage:

- https://github.com/pyca/cryptography/blob/master/src/cryptography/hazmat/backends/openssl/rsa.py#L241
- https://github.com/pyca/cryptography/blob/master/src/cryptography/hazmat/backends/openssl/rsa.py#L431

### PR DESCRIPTION
**Pull request by reaperhulk**
same approach as in #2553 for 0.9.8

fixes #2182

### CODE REVIEW

**Comment by alex** [comment]
Probably needs a rebase now.

**Comment by reaperhulk** [comment]
This depends on #2629

File: .travis.yml [hunk 1]
```diff
@@ -21,6 +22,8 @@ matrix:
           env: TOXENV=py35
         - python: pypy
           env: TOXENV=pypy
+        - python: 3.5
```
Comment (alex) [inline .travis.yml]: I'd like a 2.7 as well here. I think I'm convinced it's not super important to have the entire set of 2.6-3.5.

**Comment by alex** [comment]
This can be reabsed now.

**Pull request merged**